### ISSUE TITLE
Caching Cannot Be Configured Globally

### ISSUE THREAD

**Issue by mvastola** [rubocop-hq/rubocop#7083]
So this is somewhere between a feature request and a bug report, but basically I'm running into an issue where I'm trying to run `rubocop` on an open source project I'm working on and I noticed I'm repeatedly getting the error:
```
Warning: /home is a symlink, which is not allowed.
```
This error appears to have originated in #2484 as a way to mitigate certain attacks, and is being triggered because my `/home` folder is symlinked onto another partition. (I can't just mount the partition there for unrelated reasons.) Per #3005, however, it appears possible to disable this, and there is also a `CacheRootDirectory` configuration option that would presumably allow me to set the cache directory to a non-symlinked location.

The problem, however, is that all caching configuration options -- a) the security of a user's default cache directory (i.e. `~/.cache`); b) the need for a cache directory to be in a non-default location; and c) if a cache directory is available in the first place -- is entirely dependent on the given user/computer rather than the project being worked on. However, if a project has a `.rubocop.yml` file, the per-user config file at `~/.rubocop.yml` isn't read at all, which prevents these options from being set. 

This poses an issue on any team coding endeavor because I have no way to override the caching settings (unless I remember to use command line switches) and it would be inappropriate to commit changes to suit my specific issue to the repository of a project worked on by many.

As such, I would like to propose one or more of the following fixes:

1. Remove this check if the owner of all symlinks in the cache path are either root or the user running rubocop. (There shouldn't be any harm in this. the only risk would be if the cache directory was manually set to a malicious symlink in the `.rubocop.yml` file of a project, but said project always just could set `AllowSymlinksInCacheRootDirectory` and set the cache location to the directory they'd like to overwrite.)
2. Allow rubocop defaults to be overiden by options placed in `~/.rubocop_global.yml` (or some similar name) by default, possibly with the option to disable this practice in per-project files.
3. Allow just rubocop defaults relating to cache settings to be overridden by options placed in `~/.rubocop_cache.yml`

I'm personally for both 1 and 2 of the above.

**Comment by mvastola** [comment]
Keeping this feature request active.

**Comment by mvastola** [comment]
**bump**

**Comment by mvastola** [comment]
\**further activity*\*

**Comment by mildred** [comment]
I have this problem too, this is because I'm using Fedora Silverblue as my desktop and /home happens to be a symlink to /var/home (because / is read-only)

**Comment by duncan-bayne** [comment]
*bump*

![fist-bump](https://user-images.githubusercontent.com/65949/224172415-ff857daa-b998-446c-a38d-1cc97476d1ed.gif)

I ran into this on FreeBSD 13.1 recently.

**Comment by Math2** [comment]
I have the same problem because my ~/.cache is symlinked to a different filesystem.

One workaround is to add this to ~/.profile:
```
export RUBOCOP_CACHE_ROOT="$(realpath ~/.cache/rubocop_cache)"
```

**Comment by Earlopain** [comment]
I was looking into the context behind this option and wonder if this is even relevant anymore. Things have changed since it was added/reported:
* RuboCop uses json instead of marshal to store it's cached data (done with the initial implementation)
* The default cache directory is no longer `/tmp` but rather `$HOME/.cache` https://github.com/rubocop/rubocop/pull/4350

So it looks to me the concern that this is supposed to protect against doesn't exist anymore. `$HOME` is _not_ world-writable, so anyone who puts a symlink there can just create/delete the files they want without a middleman.
